### DISCUSSION TITLE
ZB-GW03-V1.2 with Zigbee and KNX support?

### DISCUSSION THREAD

**Discussion by gr6803** [arendst/Tasmota#18246] [Q&A]
Hi,
I purchased a wired Zigbee Bridge ZB-GW03-V1.2 with Tasmota pre-installed. I intend to use this device within a KNX installation. So I need a tasmota32 firmware, that supports ZigbeeBridge AND KNX.
Is this possible? I played around with several versions an what I got is ZigbeeBridge OR KNX.
On wireless devices with 8266 architecture, like Sonoff ZBBridge, this works.
Thanks for your support.
Gerald

**Comment by s-hadinger** [comment]
I'm not familiar with KNX but I don't see why it wouldn't work. You need to enable the required compile options. What did you try?

**Comment by Jason2866** [comment]
You can see what is used for the "unofficial" precompiled zigbeebridge32 firmware
https://github.com/Jason2866/Tasmota-build/blob/special_builder/tasmota/user_config_override.h#L750
uncomment the define for KNX

**Comment by gr6803** [comment]
Thank you for your reply. I‘m not familiar with compiling, so I tried loading the precompiled versions tasmota32-zigbeebridge and tasmota32-knx. I was looking for a precompiled file, which containes both functionalities. Maybe this is not available.RgdsGerald Vom iPhone gesendetAm 23.03.2023 um 22:42 schrieb s-hadinger ***@***.***>:﻿
I'm not familiar with KNX but I don't see why it wouldn't work. You need to enable the required compile options. What did you try?

**Reply by sfromis** [reply]
There is no precompiled binary with that unusual combination, hence what s-hadinger wrote about what you need to do.

**Reply by gr6803** [reply]
Is this really an unusual combination? What I want to do is to transfer sensor values from Zigbee devices into the KNX world. Using wireless bridges this works fine.
Meanwhile I played around with Gitpod and platformio. I was successful in compiling a simple firmware image using KNX.
Can anybody please give me a hint, how to combine the custom_config_zigbee_bridge32 section from the user_config_override.h with use_knx on tasmota32 including the EZSP? What ist the correct platformio command an what files do I have to modify in order to get the expected version?

**Reply by Jason2866** [reply]
See my answer https://github.com/arendst/Tasmota/discussions/18246#discussioncomment-5411727

**Reply by gr6803** [reply]
Yes, I saw your answer - thank you. I discovered the mentioned section in the file, I uncommented the **#Define USE_KNX** line and I put the modified file (with complete content) in my gitpod workspace. Nevertheless I do not know, how to compile in order to achieve the correct result.
I used **platformio run -e tasmota32** and got a version without zigbee.

**Reply by Jason2866** [reply]
remove the `#ifdef CUSTOM_CONFIG_ZIGBEE_BRIDGE32 ` if you compile just one firmware
So you can use the tasmota32 env. Next run of `pio run -e tasmota32` will build the zigbee firmware with KNX support.

**Reply by gr6803** [reply]
Thanks, I did so an got a working ZigbeeBridge image without KNX (no KNX in user interface and non-working KNX commands). Is there possibly another place, where KNX has to be enabled? For example in **my_user_config.h** i discovered **#define KNX_ENABLED false.**
Will this be overwritten by #DEFINE USE_KNX?

**Reply by sfromis** [reply]
You do not *need* to worry about that `#define KNX_ENABLED false`
If you have KNX included in the binary, you have the `Knx_Enabled` command to enable KNX support after restarting. See:
https://tasmota.github.io/docs/Commands/#knx

**Reply by gr6803** [reply]
I'm not sure, that KNX is really included. The KNX_Enable 1 command gives me a "RESULT = {"Command":"Unknown"}" back.

**Reply by sfromis** [reply]
You may be about that. If you have KNX support included, the "Configuration" menu should have an item "Configure KNX", where you can enable KNX. The `Knx_Enabled` command seems to only become available after having done that.

**Reply by gr6803** [reply]
Surprise, surprise! I just compiled a German language version tasmota32-DE (without changing anything) - and this image contains KNX support...

**Reply by gr6803** [reply]
I tried again today with English version and got a working image. No idea, what the difference is.
Thank you for your support.
Gerald
